### DISCUSSION TITLE
Gmail to CWP(dovecot)

### DISCUSSION THREAD

**Discussion by vandi90** [imapsync/imapsync#463] [Q&A]
I'm trying to sync from google workspace to centos web panel.  I'm coming up with issue where messages inside folders are not consistent.

Sample:
Host1 folder 117/133 [_2020]                             Size: 336313673 Messages:  3539 Biggest:  21623850
Host2 folder 117/133 [_2020]                             Size:  59821671 Messages:   693 Biggest:   8955951
Host2-Host1                                                    -276492002           -2846          -12667899

And a few folders where a single messges isn't getting moved:
Host1 folder 131/133 [temp]                              Size:  10258438 Messages:     1 Biggest:  10258438
Host2 folder 131/133 [temp]                              Size:         0 Messages:     0 Biggest:         0
Host2-Host1                                                    -10258438              -1          -10258438

I do not see a log entry for any reason why these folders shouldn't be in sync.. duplicate,too big, etc.

Thanks!

**Comment by gilleslamiral** [comment]
What gives:

    imapsync ... --folder temp

**Reply by vandi90** [reply]
If I specify the --folder option it will copy over with no issues.  I've done that on a couple that did not initially copy over, but when I leave the --folder off it still isn't picking up on the messages.  I've tried this over a couple of days so it doesn't appear to be hiting daily transfer quotas.
------

`Checking SEARCH ALL works on both accounts. To avoid that check, use --nochecknoabletosearch
Host1: checking if SEARCH ALL works on INBOX
Host1: folder [INBOX] has 7218 messages mentioned by SELECT
Host1: folder [INBOX] has 7218 messages found by SEARCH ALL
Host1: folder [INBOX] has the same messages count (7218) by SELECT and SEARCH ALL
Host2: checking if SEARCH ALL works on INBOX
Host2: folder [INBOX] has 7172 messages mentioned by SELECT
Host2: folder [INBOX] has 7172 messages found by SEARCH ALL
Host2: folder [INBOX] has the same messages count (7172) by SELECT and SEARCH ALL
Good! SEARCH ALL works on both accounts.

Folders sizes before the synchronization. It can take some time. Be patient.
You can remove foldersizes listings by using "--nofoldersizes" and "--nofoldersizesatend"
but then you will also lose the ETA (Estimation Time of Arrival) given after each message copy.
Host1 folder     1/1 [temp]                              Size:  10258438 Messages:     1 Biggest:  10258438
Host2 folder     1/1 [temp]                              Size:         0 Messages:     0 Biggest:         0
Host2-Host1                                                    -10258438              -1          -10258438

Host1 Nb folders:                     1 folders
Host2 Nb folders:                     1 folders

Host1 Nb messages:                    1 messages
Host2 Nb messages:                    0 messages

Host1 Total size:              10258438 bytes (9.783 MiB)
Host2 Total size:                     0 bytes (0.000 KiB)

Host1 Biggest message:         10258438 bytes (9.783 MiB)
Host2 Biggest message:                0 bytes (0.000 KiB)

Time spent on sizing:         4.3 seconds
++++ Looping on each one of 1 folders to sync
ETA: Friday 21 June 2024-06-21 10:36:00 -0400 EDT  0 s  1/1 msgs left
Folder     1/1 [temp]                              -> [temp]
Host1: folder [temp] has 1 messages in total (mentioned by SELECT)
Host2: folder [temp] has 0 messages in total (mentioned by SELECT)
Host2: folder [temp] permanentflags:
Host1: folder [temp] considering 1 messages
Host2: folder [temp] considering 0 messages
Host1: folder [temp] selected 1 messages, duplicates 0
Host2: folder [temp] selected 0 messages, duplicates 0
msg temp/2 {10258438}        copied to temp/1          0.27 msgs/s  2.669 MiB/s 9.783 MiB copied ETA: Friday 21 June 2024-06-21 10:36:04 -0400 EDT  0 s  0/1 msgs left
ETA: Friday 21 June 2024-06-21 10:36:04 -0400 EDT  0 s  0/1 msgs left
++++ End looping on each folder

Folders sizes after the synchronization.
You can remove this foldersizes listing by using  "--nofoldersizesatend"
Host1 folder     1/1 [temp]                              Size:  10258438 Messages:     1 Biggest:  10258438
Host2 folder     1/1 [temp]                              Size:  10258438 Messages:     1 Biggest:  10258438
Host2-Host1                                                            0               0                  0

Host1 Nb folders:                     1 folders
Host2 Nb folders:                     1 folders

Host1 Nb messages:                    1 messages
Host2 Nb messages:                    1 messages

Host1 Total size:              10258438 bytes (9.783 MiB)
Host2 Total size:              10258438 bytes (9.783 MiB)

Host1 Biggest message:         10258438 bytes (9.783 MiB)
Host2 Biggest message:         10258438 bytes (9.783 MiB)

Time spent on sizing:         0.5 seconds
++++ Statistics
Transfer started on                     : Friday 21 June 2024-06-21 10:35:56 -0400 EDT
Transfer ended on                       : Friday 21 June 2024-06-21 10:36:04 -0400 EDT
Transfer time                           : 8.4 sec
Folders synced                          : 1/1 synced
Messages transferred                    : 1
Messages skipped                        : 0
Messages found duplicate on host1       : 0
Messages found duplicate on host2       : 0
Messages found crossduplicate on host2  : 0
Messages void (noheader) on host1       : 0
Messages void (noheader) on host2       : 0
Messages found in host1 not in host2    : 0 messages
Messages found in host2 not in host1    : 0 messages
Messages deleted on host1               : 0
Messages deleted on host2               : 0
Total bytes transferred                 : 10258438 (9.783 MiB)
Total bytes skipped                     : 0 (0.000 KiB)
Message rate                            : 0.1 messages/s
Average bandwidth rate                  : 1190.4 KiB/s
Reconnections to host1                  : 0
Reconnections to host2                  : 0
Memory consumption at the end           : 200.0 MiB (*time 0.5 MiB*h) (started with 162.4 MiB)
Load end is                             : 0.00 0.00 0.00 1/442 on 1 cores
CPU time and %cpu                       : 2.24 sec 26.6 %cpu 26.6 %allcpus
Biggest message transferred             : 10258438 bytes (9.783 MiB)
Memory/biggest message ratio            : 20.4
Start difference host2 - host1          : -1 messages, -10258438 bytes (-9.783 MiB)
Final difference host2 - host1          : 0 messages, 0 bytes (0.000 KiB)
The sync looks good, all 1 identified messages in host1 are on host2.
There is no unidentified message on host1.
The sync is strict, all 1 identified messages in host2 are on host1.
Detected 0 errors
Check if a new imapsync release is available by adding --releasecheck
Homepage: https://imapsync.lamiral.info/
Exiting with return value 0 (EX_OK: successful termination) 0/50 nb_errors/max_errors PID 778879
Removing pidfile /tmp/imapsync.pid
`

**Comment by gilleslamiral** [comment]
Maybe you use `--gmail1` and `--gmail1` implies `--skipcrossduplicates`
https://imapsync.lamiral.info/FAQ.d/FAQ.Gmail.txt
...
```
--skipcrossduplicates is optional but it can save Gigabytes of hard
disk memory. Within imap protocol, Gmail presents Gmail labels as
folders, so a message labelled "Work" "ProjectX" "Urgent" ends up in
three different imap folders "Work" "ProjectX" and "Urgent" after an
imap sync. Option --skipcrossduplicates prevent this behavior.
```

Add `--noskipcrossduplicates` to avoid it

**Reply by David-Garcia9919** [reply]
I ran into a similar issue when migrating from Google Workspace to a Dovecot-based mail server. I ended up using the EdbMails [IMAP Migration Tool](https://www.edbmails.com/pages/imap-migration-tool.html), which worked pretty well for me.

It has an IMAP to IMAP migration feature that preserves folder structure, flags, and message metadata during the migration. What I liked is that it gives clear logs and can handle large or tricky messages without silently skipping them.

Might be worth giving it a try on one mailbox to see if it helps with the folder inconsistencies you’re noticing.
